### ISSUE TITLE
Error after upgrade 15.5 -> 15.6: LegacySitePackage failed for package [FSharpPackage]

### ISSUE THREAD

**Issue by chillitom** [Microsoft/visualfsharp#4438]
Received an error on startup on VS 2017 15.6 telling me that the FSharp components has failed to load and to look in the activity log where I see the below errors:

```xml
  <entry>
    <record>2036</record>
    <time>2018/03/06 08:56:32.459</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>LegacySitePackage failed for package [FSharpPackage]Source: &apos;Microsoft.VisualStudio.Composition&apos; Description: Unable to cast object of type &apos;Microsoft.VisualStudio.FSharp.Editor.Settings&apos; to type &apos;ISettings&apos;.&#x000D;&#x000A;System.InvalidCastException: Unable to cast object of type &apos;Microsoft.VisualStudio.FSharp.Editor.Settings&apos; to type &apos;ISettings&apos;.&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.CastValueTo[T](Object value)&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.&lt;&gt;c__DisplayClass70_0`2.&lt;GetExports&gt;b__1()&#x000D;&#x000A;   at System.Lazy`1.CreateValue()&#x000D;&#x000A;   at System.Lazy`1.LazyInitValue()&#x000D;&#x000A;   at System.Lazy`1.get_Value()&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.GetExportedValue[T]()&#x000D;&#x000A;   at Microsoft.VisualStudio.ComponentModelHost.ComponentModel.GetService[T]()&#x000D;&#x000A;   at Microsoft.VisualStudio.FSharp.Editor.FSharpPackage.Initialize()&#x000D;&#x000A;   at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.SetSite(IServiceProvider sp)</description>
    <guid>{871D2A70-12A2-4E42-9440-425DD92A4116}</guid>
    <hr>80004002 - E_NOINTERFACE</hr>
    <errorinfo></errorinfo>
  </entry>
  <entry>
    <record>2037</record>
    <time>2018/03/06 08:56:32.463</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>SetSite failed for package [FSharpPackage](null)</description>
    <guid>{871D2A70-12A2-4E42-9440-425DD92A4116}</guid>
    <hr>80004002 - E_NOINTERFACE</hr>
    <errorinfo></errorinfo>
  </entry>
  <entry>
    <record>2038</record>
    <time>2018/03/06 08:56:32.470</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>End package load [FSharpPackage]</description>
    <guid>{871D2A70-12A2-4E42-9440-425DD92A4116}</guid>
    <hr>80004002 - E_NOINTERFACE</hr>
    <errorinfo></errorinfo>
  </entry>
```

FSharp support isn't functioning correctly after the upgrade, no syntax highlighting or language support in the editors.

**Comment by chillitom** [comment]
Performing a _Repair_ and rebooting appears to fix the issue.

**Comment by cartermp** [comment]
@chillitom Glad to hear repair fixed this. Was this with nightlies installed?

**Comment by chillitom** [comment]
Yes, it's likely they were installed.  The extension said it was _[Experimental]_ though it still says this after the repair with version 15.6.

**Comment by cartermp** [comment]
The extension will say that regardless of nightlies (older VS infrastructure we're tying into, you'll noticed that Roslyn is the same). That said, if you had a nightly installed then it may be the reason why you've seen this:

* Nightly builds, until yesterday afternoon PST, were based on **15.5**
* We just merged our 15.6 branch into master to kick off a new nightly build, which should now requure **15.6**
* Upgrading during this time may be a bit funky

Unfortunately, there isn't a good way to upgrade when they're installed 😢 

I'll close the issue for now; please do ping if you see something like this again and we can open it to investigate

**Comment by realvictorprm** [comment]
Just running into this problem too as I merged dev15.8 branch into my code lens prior. I guess it requires a newer Visual Studio version...

Going back to merge in latest master hoping that it disapear.
